sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -113,6 +114,8 @@ class SparkSession private(
 
   private[sql] val sessionUUID: String = UUID.randomUUID.toString
 
+  private[sql] val jobArtifactState: JobArtifactState = JobArtifactState(sessionUUID, None)
```

Comment:
@HyukjinKwon the code is here, add jobArtifactState to spark session

Reply:
We should make sure this works fine with Spark Connect. Now we're setting `session.jobArtifactState` at `withActive`, and it will overwrite session-specific jobArtifactState set in Spark Connect server.

Reply:
yes, this is a problem, how about update session default jobArtifactState sametime when create jobArtifactState in spark connect

Reply:
another way is maintain jobArtifactState in spark session only,  anywhere need it can get it from spark session, but this will change too many code (i think)